spec/presenters/step_by_step_presenter_spec.rb

```diff
@@ -0,0 +1,74 @@
+require "rails_helper"
+require "gds_api/test_helpers/link_checker_api"
+
+RSpec.describe StepByStepPresenter do
+  include GdsApi::TestHelpers::LinkCheckerApi
+  include AuthenticationControllerHelpers
+  include GovukContentSchemaTestHelpers
+  include TimeOptionsHelper
+
+  before do
+    allow(Services.publishing_api).to receive(:lookup_content_id)
+    stub_user.name = "Firstname Lastname"
+  end
+
+  describe "#summary_list" do
+    let(:step_nav) { create(:step_by_step_page_with_steps) }
+    let(:default_summary) {
+      [
+        {
+          field: "Status",
+          value: "Draft"
+        },
+        {
+          field: "Last saved",
+          value: format_full_date_and_time(Time.zone.now)
+        },
+        {
+          field: "Created",
+          value: format_full_date_and_time(Time.zone.now)
+        }
+      ]
+    }
+
+    subject { described_class.new(step_nav).summary_list }
+
+    it "presents a list of key/value metadata" do
+      expect(subject).to eq(default_summary)
+    end
+
+    context "assigned_to is set" do
+      let(:step_nav) { create(:step_by_step_page_with_navigation_rules, assigned_to: stub_user.name) }
+
+      it "shows who made the most recent change" do
+        expect(subject[1]).to eq(
+          field: "Last saved",
+          value: "#{format_full_date_and_time(Time.zone.now)} by #{stub_user.name}"
+        )
+      end
+    end
+
+    context "links have been checked" do
+      before :each do
+        stub_link_checker_api_create_batch(
```

Comment:
I don't think link-checker-api needs to be stubbed. I think you should be able to create a `link-report` factory, and link it to a step.

For example:
```
step = create(:step, step_by_step_page: step_nav)
create(link_report, step: step)
```

Reply:
Thanks, much nicer!